sdks/js/ditto-chat-ui/src/DittoChatUI.tsx

```diff
@@ -168,13 +170,69 @@ export default function DittoChatUI({
       mediaQueryList.addEventListener('change', handleChange)
     } else {
       mediaQueryList.removeEventListener('change', handleChange)
+      if (typeof theme === 'object') {
+        setThemeName(theme.variant || 'light')
+      } else {
+        setThemeName(theme)
+      }
     }
     // This is the cleanup function that will be called when the component unmounts
     return () => {
       mediaQueryList.removeEventListener('change', handleChange)
     }
   }, [theme])
 
+  const themeStyles = useMemo(() => {
+    if (typeof theme !== 'object') {
+      return {}
+    }
+
+    const styles: React.CSSProperties = {}
+    const mapping: Record<string, string> = {
+      primaryColor: '--dc-primary-color',
+      primaryColorHover: '--dc-primary-color-hover',
+      primaryColorFocus: '--dc-primary-color-focus',
+      primaryColorLight: '--dc-primary-color-light',
+      primaryColorLighter: '--dc-primary-color-lighter',
+      primaryColorLightBorder: '--dc-primary-color-light-border',
+      primaryColorDarkText: '--dc-primary-color-dark-text',
+      textOnPrimary: '--dc-text-on-primary',
+      mentionText: '--dc-mention-text',
+      mentionTextOnPrimary: '--dc-mention-text-on-primary',
+      surfaceColor: '--dc-surface-color',
+      surfaceColorLight: '--dc-surface-color-light',
+      secondaryBg: '--dc-secondary-bg',
+      secondaryBgHover: '--dc-secondary-bg-hover',
+      disabledBg: '--dc-disabled-bg',
+      textColor: '--dc-text-color',
+      textColorMedium: '--dc-text-color-medium',
+      textColorLight: '--dc-text-color-light',
+      textColorLighter: '--dc-text-color-lighter',
+      textColorLightest: '--dc-text-color-lightest',
+      textColorFaint: '--dc-text-color-faint',
+      textColorDisabled: '--dc-text-color-disabled',
+      borderColor: '--dc-border-color',
+      editBg: '--dc-edit-bg',
+      editText: '--dc-edit-text',
+      infoIconColor: '--dc-info-icon-color',
+      notificationBadgeBg: '--dc-notification-badge-bg',
+      activeStatusBg: '--dc-active-status-bg',
+      dangerText: '--dc-danger-text',
+      dangerBg: '--dc-danger-bg',
+      successBg: '--dc-success-bg',
+      successText: '--dc-success-text',
+    }
+
+    Object.keys(theme).forEach((key) => {
+      if (key in mapping) {
+        // @ts-expect-error - key is a valid key of Theme
+        styles[mapping[key]] = theme[key as keyof Theme]
+      }
+    })
+
+    return styles
+  }, [theme])
+
   useEffect(() => {
     localStorage.setItem('ditto-web-chat-theme', themeName)
```

Comment:
I think we can probably remove or adjust this effect? We probably _want_ the theme to persist in a way so the first paint looks correct, so maybe we save the entire theme and remove the cleanup on the effect? What do you think?